src/app/components/accountRow/index.tsx

```diff
@@ -199,6 +205,14 @@ function AccountRow({
   >();
   const { removeLedgerAccount, renameAccount, updateLedgerAccounts } = useWalletReducer();
 
+  const totalBalance = accountBalances[account?.btcAddress ?? ''];
+
+  useEffect(() => {
+    if (fetchBalance && shouldFetch && !totalBalance) {
+      fetchBalance(account);
+    }
+  }, [shouldFetch, totalBalance]);
```

Comment:
how about we move this logic to the parent component? we are making a decision in the child that the parents can make.